src/main/java/mServer/crawler/AddToFilmlist.java

```diff
@@ -5,176 +5,270 @@
+     */
+    private final ArrayList<ImportOldFilmlistThread> threadList = new ArrayList<>();
+    private AtomicInteger threadCounter = new AtomicInteger(0);
+    
 
     public AddToFilmlist(ListeFilme vonListe, ListeFilme listeEinsortieren) {
         this.vonListe = vonListe;
         this.listeEinsortieren = listeEinsortieren;
     }
 
     public synchronized void addLiveStream() {
-        EtmPoint performancePoint = etmMonitor.createPoint("AddToFilmlist:addLiveStream");
-        // live-streams einfügen, es werde die vorhandenen ersetzt!
-
-        if (listeEinsortieren.size() <= 0) {
-            //dann wars wohl nix
-            return;
-        }
+        if (listeEinsortieren.size() <= 0) return;
 
-        Iterator<DatenFilm> it = vonListe.iterator();
-        while (it.hasNext()) {
-            DatenFilm f = it.next();
-            if (f.arr[DatenFilm.FILM_THEMA].equals(ListeFilme.THEMA_LIVE)) {
-                it.remove();
-            }
-        }
+        vonListe.removeIf(f -> f.arr[DatenFilm.FILM_THEMA].equals(ListeFilme.THEMA_LIVE));
         listeEinsortieren.forEach(vonListe::add);
-        performancePoint.collect();
     }
 
-    public synchronized int addOldList() {
-        // in eine vorhandene Liste soll eine andere Filmliste einsortiert werden
-        // es werden nur Filme die noch nicht vorhanden sind, einsortiert
-        threadCounter = new AtomicInteger(0);
-        treffer = new AtomicInteger(0);
-        int size = listeEinsortieren.size();
-        HashSet<String> hash = new HashSet<>(listeEinsortieren.size() + 1, 1);
-
-        Duration.staticPing("AddOld-1");
+    private void performTitleSearch(HashSet<Hash> hash, final int size) {
+        vonListe.parallelStream().forEach(f -> {
+            synchronized (hash) {
+                hash.add(f.getHashValueIndexAddOld());
+            }
+        });
 
-        // ==============================================
-        // nach "Thema-Titel" suchen
-        EtmPoint performancePointThemaTitel = etmMonitor.createPoint("AddToFilmlist:addOldList#themaTitel");
-        vonListe.stream().forEach((f) -> hash.add(f.getIndexAddOld()));
-        listeEinsortieren.removeIf((f) -> hash.contains(f.getIndexAddOld()));
+        listeEinsortieren.removeIf((f) -> hash.contains(f.getHashValueIndexAddOld()));
         hash.clear();
-        performancePointThemaTitel.collect();
 
-        Log.sysLog("===== Liste einsortieren Hash =====");
+        Log.sysLog("===== Liste einsortieren Title =====");
         Log.sysLog("Liste einsortieren, Anzahl: " + size);
         Log.sysLog("Liste einsortieren, entfernt: " + (size - listeEinsortieren.size()));
         Log.sysLog("Liste einsortieren, noch einsortieren: " + listeEinsortieren.size());
         Log.sysLog("");
-        size = listeEinsortieren.size();
+    }
+
+    private void performUrlSearch(HashSet<Hash> hash, final int size) {
+        vonListe.parallelStream().forEach(f -> {
+            synchronized (hash) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') "hash" is a method parameter, and should not be used for synchronization. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2445)